Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/HistoryEntityRequest.java

```diff
@@ -123,8 +123,7 @@ protected JsonElement setCommonTemporalProperties(JsonElement jsonElement, Strin
 			valueCreatedAt = date;
 		}
 		// append/overwrite temporal fields. as we are creating new instances,
-		// modifiedAt and createdAt are the same
-		jsonElement = setTemporalProperty(jsonElement, NGSIConstants.NGSI_LD_CREATED_AT, valueCreatedAt);
+		// modifiedAt and createdAt are the same		
```

Comment:
In case of append it will not work because createdAt and modifiedAt will be same so please check again.

Reply:
I have checked. It is working on both append and update.

Reply:
Hi please validate if createdAt still gets set on a create of a temporal entity, i fear not. This all seems a bit weird from the logic. It looks a bit if this code was intended to be for creates only where modifiedat and createdat are the same.
Seeing how update, create and append request it needs a bit of code changes.
so at the moment we check if there is a modifiedAt in there already (as this comes from an entity) ... 
1. rename valueCreatedAt to valueModifiedAt as it is what is gotten there.
2. do the same check as in line 109 to 120 for createdAt to and see if it already exists.
3. as this method is used by all 3 variants add a boolean to indicate if create should be added.
4. check if append works properly in the sense that an append could have an existing attribute and a new one in the same request